subprojects/docs/src/docs/userguide/running-builds/gradle_daemon.adoc

```diff
@@ -258,9 +258,53 @@ This message means that either
 --
 In both cases the build cannot benefit from file system watching.
 
-`Caught exception: Couldn't add watch, error = 28`::
-If you receive this error on Linux then you ran out of inotify handles.
-To raise the limit see https://confluence.jetbrains.com/display/IDEADEV/Inotify+Watches+Limit[here].
-
 `java.io.IOException: Too many open files`::
 If you receive this error on macOS, you need to raise your open files limit, see https://superuser.com/a/443168/8117[here].
+
+===  Linux specific notes
+
+File system watching uses http://en.wikipedia.org/wiki/Inotify[Inotify] on Linux.
+Depending on the size of your build, it may be necessary to increase Inotify limits.
+If you are using an IDE, then you probably already did increase the limits.
+
+[[sec:inotify_watches_limit]]
+Inotify watches limit::
+File system watching uses one Inotify watch per watched directory.
+You can see the current limit of Inotify watches per user by running:
++
+[source,bash]
+----
+cat /proc/sys/fs/inotify/max_user_watches
+----
++
+To increase the limit to e.g. 512K watches:
```

Comment:
I think we should address the memory impact of doing this. This here says the cost of each watch is around 1K of memory: https://blog.passcod.name/2017/jun/25/inotify-watch-limit#:~:text=An%20INotify%20watch%20costs%20540,this%20time%20in%20swappable%20memory).

However, IIUC only the actually used watches actually consume memory: https://www.man7.org/linux/man-pages//man7/inotify.7.html#:~:text=/proc%20interfaces

Reply:
I added some information about the memory impact.